File: server/coffeeshop/src/main/java/com/hust/coffeeshop/services/ComboService.java
package com.hust.coffeeshop.services;

import com.hust.coffeeshop.models.dto.PagingListResponse;
import com.hust.coffeeshop.models.dto.combo.request.CreateComboRequest;
import com.hust.coffeeshop.models.dto.combo.response.ComboRespone;
import com.hust.coffeeshop.models.dto.item.request.ItemRequest;
import com.hust.coffeeshop.models.entity.ComboItem;

import java.io.IOException;
import java.util.List;

public interface ComboService {
    ComboRespone create(CreateComboRequest request) throws IOException;
    ComboRespone update(CreateComboRequest request,int id);
    ComboRespone getbyId(int id);
    void delete(int id);
    PagingListResponse<ComboRespone> filter(ItemRequest filter);

    List<ComboItem> getComboItemByComboIds(List<Integer> comboId);
}

File: client/coffee-shop/src/services/ComboService/ComboService.ts
import axios, { AxiosResponse } from "axios";
import { getAxiosConfig } from "../config";
import { ComboFilterRequest, ComboRespone, CreateComboRequest, ListComboRespone } from "./types";

class ComboService {
    static async filter(filter: ComboFilterRequest): Promise<AxiosResponse<ListComboRespone>> {
        return axios.get(`/combo`, { ...getAxiosConfig(), params: filter});
    }
    static async create(request: CreateComboRequest): Promise<AxiosResponse<ComboRespone>> {
        return axios.post(`/combo`,request , { ...getAxiosConfig()});
    }
    static async update(request: CreateComboRequest, id: string): Promise<AxiosResponse<ComboRespone>> {
        return axios.put(`/combo/${id}`,request, { ...getAxiosConfig()});
    }
    static async getById(id: string): Promise<AxiosResponse<ComboRespone>> {
        return axios.get(`/combo/${id}`, { ...getAxiosConfig()});
    }
    static async delete(id: string): Promise<AxiosResponse<ComboRespone>> {
        return axios.delete(`/combo/${id}`, { ...getAxiosConfig()});
    }
}
export default ComboService;